src/ethereum/EthereumNetwork.ts

```diff
@@ -1537,7 +1537,13 @@ export class EthereumNetwork {
 
     let funcs: Array<() => Promise<any>> = []
     rpcServers.forEach(rpcServer => {
-      const rpcServerWithApiKey = this.addRpcApiKey(rpcServer)
+      let rpcServerWithApiKey: string
+      try {
+        rpcServerWithApiKey = this.addRpcApiKey(rpcServer)
+      } catch (e) {
+        // addRpcApiKey can throw if there's a missing apikey. skip this rpcServer
+        return
+      }
```

Comment:
Hmm, if all RPC servers fail on this adding of key step, then this means balance checking via RPC is not possible. There are other methods to get balances which wont be used because of the condition on line 1787. This can surface itself as a bug as well, so this only partly fixes the problem.